README.md
# Zurb Foundation for Jekyll

Zurb Foundation for Jekyll (namespace: zfjk), is a blank starter site built using version 4.x of [ZURB's Foundation Framework](http://foundation.zurb.com/), the [Jekyll](http://jekyllrb.com/) static site generator and [Grunt](http://gruntjs.com/) for build/deploy automation.

It features...and lot's of other good stuff.

## Download

Clone the git repo - `git clone https://github.com/jhauraw/foundation-jekyll.git` - or, [download the archive](https://github.com/jhauraw/foundation-jekyll/zipball/master).

## Author

**Jhaura Wachsman**

Integrated Foundation into Jekyll, develope and build automation with Grunt and LiveReload.

+ [Twitter](http://twitter.com/JhauraWachsman)
+ [Blog](http://jhauarawachsman.com)

## Framework

**JEKYLL**

+ The Jekyll static site generator was created by Tom Preston-Werner, a co-founder at GitHub.

**ZURB**

+ The Foundation framework was made by ZURB, an interaction and design strategy firm in California, USA.

CHANGELOG.md
TODO:
  LINK ASSETS! GET ASSETS PARSED, COPIED AND MIN'D

  flatterline jekyll github source for their blog
  blitz.io 0-250 in 10s, then sustain 250 for 50s



Gruntfile.js
/*global module:false*/
module.exports = function(grunt) {

  grunt.initConfig({

    pkg: grunt.file.readJSON('package.json'),

    banner: '/*! <%= pkg.title || pkg.name %> - v<%= pkg.version %> - ' +
    '<%= grunt.template.today("yyyy-mm-dd") %>\n' +
    '<%= pkg.homepage ? "* " + pkg.homepage + "\\n" : "" %>' +
    '* Copyright (c) <%= grunt.template.today("yyyy") %> <%= pkg.author.name %>;' +
    ' Licensed <%= _.pluck(pkg.licenses, "type").join(", ") %> */\n',

    compass: {
      dist: {
        options: {
          require: 'zurb-foundation',
          sassDir: '_assets/sass',
          cssDir: 'css',
          outputStyle: 'expanded',
          noLineComments: true,
          app: 'standalone',
          environment: 'production',
          raw: 'preferred_syntax = :scss\n'
        }
      }
    },
    cssmin: {
      compress: {
        files: {
          'css/app.min.css': ['css/site.css']
        }
      }
    },
    concat: {
      options: {
        banner: '<%= banner %>',
        stripBanners: true
      },
      dist: {
        src: ['lib/<%= pkg.name %>.js'],
        dest: 'dist/<%= pkg.name %>.js'
      }
    },
    'jshint': {
      all: ['js/*.js', '!js/modernizr.js', '!js/foundation.js', '!js/jquery.js', '!js/zepto.js', '!js/vendor/**/*.js'],
      options: {
        browser: true,
        curly: false,
        eqeqeq: false,
        eqnull: true,
        expr: true,
        immed: true,
        newcap: true,
        noarg: true,
        smarttabs: true,
        sub: true,
        undef: false
      }
    },
    uglify: {
      options: {
        banner: '<%= banner %>'
      },
      dist: {
        src: '<%= concat.dist.dest %>',
        dest: 'dist/app.min.js'
      }
    },
    regarde: {
      js: {
        files: '_assets/js/**/*.js',
        tasks: ['jshint', 'livereload'],
        //spawn: true
      },
      css: {
        files: '_assets/sass/**/*.scss',
        tasks: ['compass:dist', 'livereload'],
        events: true
      },
      php: {
        files : '**/*.php',
        tasks : 'livereload'
      }
    },

    gruntfile: {
      src: 'Gruntfile.js'
    }
  });

  grunt.registerTask('default', ['livereload-start', 'regarde']);
  grunt.registerTask('dist', ['compass', 'cssmin', 'jshint', 'uglify', 'concat']);

  grunt.loadNpmTasks('grunt-regarde');
  grunt.loadNpmTasks('grunt-contrib-livereload');

  grunt.loadNpmTasks('grunt-contrib-concat');

  grunt.loadNpmTasks('grunt-contrib-compass');
  grunt.loadNpmTasks('grunt-contrib-cssmin');

  grunt.loadNpmTasks('grunt-contrib-jshint');
  grunt.loadNpmTasks('grunt-contrib-uglify');
};


_assets/js/app/_global.js
;(function ($, window, document, undefined) {
  'use strict';

  Foundation.libs.navmenu = {
    name : 'navmenu',

    init : function (scope, method, options) {

      var wp = $('.navsection'),
          ul = $('.navmenu'),
          li = $('.navmenu>li.submenu'),
          content,
          state;

      $(li, this).on('mouseenter', function (e) {

        content = $(this).children('.submenu-content');

        state = ul.find('.submenu.active, .submenu.locked').length > 0;

        ul.find('.submenu').removeClass('active locked');
        $(this).addClass('active locked');

        if (!state) {

          if (window.Zepto) {

            content.animate({ translate: '0, -300px' }, 0);
          }
          else {

            content.css({'transform': 'translate(0, -300px)'});
          }

          setTimeout(function () {

            if (window.Zepto) {

              content.animate({ translate: '0, 0' }, 290);
            }
            else {

              content.css({'transition': 'all 290ms linear', 'transform': 'translate(0,0)'});
            }
          }, 0);
        }
      });

      $(wp, this).on('mouseleave', function (e) {

        wp.find('.submenu').removeClass('active locked');
      });
    }
  };
}(Foundation.zj, this, this.document));

;(function ($, window, document, undefined) {
  'use strict';

  Foundation.libs.panelexpose = {
    name : 'panelexpose',

    init : function (scope, method, options) {
      var self = this;
      this.scope = scope || this.scope;

      var meta   = $('meta[name=viewport]'),
          head   = $('html, body'),
          button = $('.toggle-menu a'),
          peek   = $('#peek'),
          sticky = $('#sticky'),
          page   = $('#scrollable'),
          panel  = $('#panelnav');

      if (window.Zepto) {

        $(this.scope)
          .on('swipeRight', page, function(e){

            if (!peek.hasClass('expose')) {

              button.trigger('click');
            }
          })

          .on('swipeLeft', page, function(e){

            if (peek.hasClass('expose')) {

              button.trigger('click');
            }
          });
      }

      button.click(function (e) {
        e.preventDefault();

        //meta.attr('content', 'width=device-width,minimum-scale=1.0');

        peek.toggleClass('expose');

        if (peek.hasClass('expose')) {

          /* Required! Otherwise, nasty horizontal scroll bar. */
          //$('html').css('overflow-x', 'hidden');
          //$('body').css('overflow-x', 'hidden');

          panel.show();

          if (window.Zepto) {

            peek.animate({translate: '250px, 0'}, 290);
          }
          else {

            peek.css({'transition': 'all 290ms linear', 'transform': 'translate(250px ,0)'});
          }

          page.click(function (e) {
            e.preventDefault();

            self.clean(head, meta, peek, sticky, page, panel, e);

          }); // end page click
        }
        else {

          self.clean(head, meta, peek, sticky, page, panel, e);

        } // end else
      }); // end button click
    },

    clean : function (head, meta, peek, sticky, page, panel, e) {
      var self = this;

      //meta.attr('content', 'width=device-width');

      if (window.Zepto) {

        peek.animate({translate: '0, 0'}, 150);
      }
      else {

        peek.css({'transition': 'all 150ms linear', 'transform': 'translate(0 ,0)'});
      }

      setTimeout(function () {

        peek
          .css('-webkit-transform', 'none')
          .css('-webkit-transform', '')
          .removeClass('expose');

        panel.hide();

        //$('html').css('overflow', 'hidden');
        //$('body').css('overflow', '');

        //$('html').css('overflow-x', 'visible');
        //$('body').css('overflow-x', 'visible');

      }, 300); // just longer than the expose
    }
  };
}(Foundation.zj, this, this.document));

;(function ($, window, document, undefined) {
  'use strict';

  Foundation.libs.iosfix = {
    name : 'iosfix',

    init : function (scope, method, options) {
      var self = this;
      this.scope = scope || this.scope;

      var sticky = $('#sticky'),
          panel  = $('#panelnav');

      if(!/(ip(hone|od|ad))/i.test(navigator.userAgent))
        return false;

      $('input[type="text"], input[type="password"], input[type="search"], textarea').css('font-size', '1em');

      $('input')
        .focus(function(e) {

          sticky.add(panel).css('position', 'absolute');
        })

        .blur(function(e) {

          //window.offset({ top: 0, left: 0 });

          //sticky.add(panel).css({'top': 0, 'position': 'fixed'});
        });

      window.addEventListener("load",function() {

        setTimeout(function () {

          window.scrollTo(0, 1);

        }, 0);
      });
    }
  };
}(Foundation.zj, this, this.document));

$(document).foundation();

_assets/js/app/app.js
;(function ($, window, document, undefined) {
  'use strict';

  Foundation.libs.navmenu = {
    name : 'navmenu',

    init : function (scope, method, options) {

      var wp = $('.navsection'),
          ul = $('.navmenu'),
          li = $('.navmenu>li.submenu'),
          content,
          state;

      $(li, this).on('mouseenter', function (e) {

        content = $(this).children('.submenu-content');

        state = ul.find('.submenu.active, .submenu.locked').length > 0;

        ul.find('.submenu').removeClass('active locked');
        $(this).addClass('active locked');

        if (!state) {

          if (window.Zepto) {

            content.animate({ translate: '0, -300px' }, 0);
          }
          else {

            content.css({'transform': 'translate(0, -300px)'});
          }

          setTimeout(function () {

            if (window.Zepto) {

              content.animate({ translate: '0, 0' }, 290);
            }
            else {

              content.css({'transition': 'all 290ms linear', 'transform': 'translate(0,0)'});
            }
          }, 0);
        }
      });

      $(wp, this).on('mouseleave', function (e) {

        wp.find('.submenu').removeClass('active locked');
      });
    }
  };
}(Foundation.zj, this, this.document));

;(function ($, window, document, undefined) {
  'use strict';

  Foundation.libs.panelexpose = {
    name : 'panelexpose',

    init : function (scope, method, options) {
      var self = this;
      this.scope = scope || this.scope;

      var meta   = $('meta[name=viewport]'),
          head   = $('html, body'),
          button = $('.toggle-menu a'),
          peek   = $('#peek'),
          sticky = $('#sticky'),
          page   = $('#scrollable'),
          panel  = $('#panelnav');

      if (window.Zepto) {

        $(this.scope)
          .on('swipeRight', page, function(e){

            if (!peek.hasClass('expose')) {

              button.trigger('click');
            }
          })

          .on('swipeLeft', page, function(e){

            if (peek.hasClass('expose')) {

              button.trigger('click');
            }
          });
      }

      button.click(function (e) {
        e.preventDefault();

        //meta.attr('content', 'width=device-width,minimum-scale=1.0');

        peek.toggleClass('expose');

        if (peek.hasClass('expose')) {

          /* Required. Otherwise, nasty horizontal scroll bar. */
          //$('html').css('overflow-x', 'hidden');
          //$('body').css('overflow-x', 'hidden');

          panel.show();

          if (window.Zepto) {

            peek.animate({translate: '250px, 0'}, 290);
          }
          else {

            peek.css({'transition': 'all 290ms linear', 'transform': 'translate(250px ,0)'});
          }

          page.click(function (e) {
            e.preventDefault();

            self.clean(head, meta, peek, sticky, page, panel, e);

          }); // end page click
        }
        else {

          self.clean(head, meta, peek, sticky, page, panel, e);

        } // end else
      }); // end button click
    },

    clean : function (head, meta, peek, sticky, page, panel, e) {
      var self = this;

      //meta.attr('content', 'width=device-width');

      if (window.Zepto) {

        peek.animate({translate: '0, 0'}, 150);
      }
      else {

        peek.css({'transition': 'all 150ms linear', 'transform': 'translate(0 ,0)'});
      }

      setTimeout(function () {

        peek
          .css('-webkit-transform', 'none')
          .css('-webkit-transform', '')
          .removeClass('expose');

        panel.hide();

        //$('html').css('overflow', 'hidden');
        //$('body').css('overflow', '');

        //$('html').css('overflow-x', 'visible');
        //$('body').css('overflow-x', 'visible');

      }, 300); // just longer than the expose
    }
  };
}(Foundation.zj, this, this.document));

;(function ($, window, document, undefined) {
  'use strict';

  Foundation.libs.iosfix = {
    name : 'iosfix',

    init : function (scope, method, options) {
      var self = this;
      this.scope = scope || this.scope;

      var sticky = $('#sticky'),
          panel  = $('#panelnav');

      if(!/(ip(hone|od|ad))/i.test(navigator.userAgent))
        return false;

      $('input[type="text"], input[type="password"], input[type="search"], textarea').css('font-size', '1em');

      $('input')
        .focus(function(e) {

          sticky.add(panel).css('position', 'absolute');
        })

        .blur(function(e) {

          //window.offset({ top: 0, left: 0 });

          //sticky.add(panel).css({'top': 0, 'position': 'fixed'});
        });

      window.addEventListener("load",function() {

        setTimeout(function () {

          window.scrollTo(0, 1);

        }, 0);
      });
    }
  };
}(Foundation.zj, this, this.document));

$(document).foundation();